### PR TITLE
User managed interrupt: add check on number of interrupt

### PR DESCRIPTION
**Pull request by sarab96**
#### Problem solved by the commit
None. Warning added for number of interrupts. Usual behavior expects only one interrupt

#### How problem was solved, alternative solutions (if any) and why they were rejected
New check added

#### Documentation impact (if any)
None

### CODE REVIEW

File: src/runtime_src/core/pcie/linux/device_linux.cpp [hunk 1]
```diff
@@ -1012,9 +1012,12 @@ void
 device_linux::
 wait_ip_interrupt(xclInterruptNotifyHandle handle)
 {
-  int pending = 0;
+  int num_interrupts = 0;
   if (::read(handle, &pending, sizeof(pending)) == -1)
     throw error(errno, "wait_ip_interrupt failed POSIX read");
+
+  if (num_interrupts > 1)
+    xrt_core::message::send(xrt_core::message::severity_level::warning, "XRT", "More than one interrupt was received");
```
Comment (stsoe) [inline src/runtime_src/core/pcie/linux/device_linux.cpp:1020]: I don't see what this PR is doing.  Don't just add unused code.

**Review by stsoe — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/pcie/linux/device_linux.cpp (hunk 1 above)
Comment (sarab96) [reply src/runtime_src/core/pcie/linux/device_linux.cpp]: Updated the code

**Review by stsoe — CHANGES_REQUESTED** [review]
I understand the change, but I don't understand the bigger picture?
What does num_interrupts represent and why is it a warning if the number is > 1?
It is important that you document the changes in code (comments) if it is not obvious.

**Comment by sarab96** [comment]
Will discuss with Soren and resubmit later. Right now this generates too many warnings with the way xrt::interrupt::wait() is implemented. As wait() automatically reenables interrupts.

**Pull request closed without merging**